android/app/src/googleplay/java/org/fossasia/openevent/utils/map/ClusterRenderer.java

```diff
@@ -0,0 +1,95 @@
+package org.fossasia.openevent.utils.map;
+
+import android.content.Context;
+import android.os.Handler;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.maps.android.clustering.ClusterManager;
+import com.google.maps.android.clustering.view.DefaultClusterRenderer;
+
+import org.fossasia.openevent.R;
+
+/**
+ * Created by rohanagarwal94 on 4/7/17.
+ */
+public class ClusterRenderer extends DefaultClusterRenderer<MicrolocationClusterWrapper> implements ClusterManager.OnClusterItemClickListener<MicrolocationClusterWrapper>, GoogleMap.OnMapClickListener {
+
+    private Context context;
+    private GoogleMap map;
+    private MicrolocationClusterWrapper microlocationClusterWrapper;
+
+    public ClusterRenderer(Context context, GoogleMap map,
+                             ClusterManager<MicrolocationClusterWrapper> clusterManager) {
+        super(context, map, clusterManager);
+        this.context = context;
+        this.map = map;
+        setOnClusterItemClickListener(this);
```

Comment:
Now, the strong reference to this class is passed to a listener, you must set it null on detach